src/ethereum_test_filling_tool/main.py

```diff
@@ -154,6 +165,92 @@ def fill(self) -> None:
                     fixture, f, ensure_ascii=False, indent=4, cls=JSONEncoder
                 )
 
+        if self.options.benchmark:
+            end_time = time.time()
+            elapsed_time = end_time - start_time
+            self.log.info(
+                f"Filled test fixtures in {elapsed_time:.2f} seconds."
+            )
+
+    def parallel_fill(self) -> None:
+        """
+        Parallel Fill test fixtures.
+        """
+        if self.options.benchmark:
+            start_time = time.time()
+
+        pkg_path = self.options.filler_path
+
+        fillers = []
+
+        for package_name, module_name, module_loader in find_modules(
+            os.path.abspath(pkg_path),
+            self.options.test_categories,
+            self.options.test_module,
+        ):
+            module_full_name = module_loader.name
+            self.log.debug(f"searching {module_full_name} for fillers")
+            module = module_loader.load_module()
+            for obj in module.__dict__.values():
+                if callable(obj):
+                    if hasattr(obj, "__filler_metadata__"):
+                        if (
+                            self.options.test_case
+                            and self.options.test_case
+                            not in obj.__filler_metadata__["name"]
+                        ):
+                            continue
+                        obj.__filler_metadata__["module_path"] = [
+                            package_name,
+                            module_name,
+                        ]
+                        fillers.append(obj)
+
+        self.log.info(f"collected {len(fillers)} fillers")
+
+        os.makedirs(self.options.output, exist_ok=True)
+
+        t8n = EvmTransitionTool(
+            binary=self.options.evm_bin, trace=self.options.traces
+        )
+        b11r = EvmBlockBuilder(binary=self.options.evm_bin)
+
+        with concurrent.futures.ThreadPoolExecutor() as executor:
```

Comment:
I think this method can be made the default one and perhaps we can include an argument to specify `max_workers` here, which could be set to `1` if the user wants to fill them serially.